### PR TITLE
Removes use of typecache in get area

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

also fixes admin mining scanner

## Changelog
:cl:
refactor: removed static usage of typecache in get_area()
/:cl:

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -608,18 +608,16 @@ Returns: A list of all areas of that type in the world.
 
 	var/list/areas = list()
 	if(subtypes)
-		var/list/cache = typecacheof(areatype)
-		for(var/V in GLOB.sortedAreas)
-			var/area/A = V
-			if(cache[A.type])
+		var/list/cache = typesof(areatype)
+		for(var/area/A as() in GLOB.sortedAreas)
+			if(A.type in cache)
```
Comment (PowerfulBacon) [inline code/__HELPERS/unsorted.dm]: Why not just do istype(A, areatype). I'm pretty sure its faster than using in

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -638,17 +636,15 @@ Returns: A list of all turfs in areas of that type of that type in the world.
 
 	var/list/turfs = list()
 	if(subtypes)
-		var/list/cache = typecacheof(areatype)
-		for(var/V in GLOB.sortedAreas)
-			var/area/A = V
-			if(!cache[A.type])
+		var/list/cache = typesof(areatype)
+		for(var/area/A as() in GLOB.sortedAreas)
+			if(!(A.type in cache))
```
Comment (PowerfulBacon) [inline code/__HELPERS/unsorted.dm]: !istype(A, areatype)

File: code/__HELPERS/unsorted.dm (hunk 1 above)
Comment (Archanial) [reply code/__HELPERS/unsorted.dm]: not exactly sure what you mean here

Comment (PowerfulBacon) [reply code/__HELPERS/unsorted.dm]: A in list is slow. Just do istype(A, areatype) and then the cache isn't needed there

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -608,18 +608,16 @@ Returns: A list of all areas of that type in the world.
 
 	var/list/areas = list()
 	if(subtypes)
-		var/list/cache = typecacheof(areatype)
-		for(var/V in GLOB.sortedAreas)
-			var/area/A = V
-			if(cache[A.type])
+		var/list/cache = typesof(areatype)
```
Comment (PowerfulBacon) [inline code/__HELPERS/unsorted.dm]: Pointsless typesof?

Comment (Archanial) [inline code/__HELPERS/unsorted.dm]:
```suggestion
```

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**